os/pm/pm_metrics.c

```diff
@@ -17,117 +17,289 @@
  ****************************************************************************/
 
 #include <tinyara/config.h>
+#include <tinyara/pm/pm.h>
 #include <time.h>
 #include <queue.h>
 #include <debug.h>
-#include "pm_metrics.h"
+#include <errno.h>
 #include "pm.h"
 
-time_t time_diff(time_t time1, time_t time2)
+struct pm_metric_domain_s {
+	uint32_t idle_suspend_counts[CONFIG_PM_NDOMAINS];   /* It counts the suspended domain inside idle thread which prevent board sleep */
+	uint32_t total_suspend_counts;                      /* It counts no. of times idle thread was not able to make board sleep */
```

Comment:
This name can be confused.
try_board_sleep_count? and total_blocking_board_sleep_count?
Please let me know better idea